### PR TITLE
docs: move wiki

### LINKED ISSUE #1314: Syntax/semantics doc pages are missing content from wiki
Historically we've had a few wiki page documenting syntax/semantics for our three DSLs, and our standard library:

- https://github.com/penrose/penrose/wiki/Language-Specification
- https://github.com/penrose/penrose/wiki/Style-function-library
- https://github.com/penrose/penrose/wiki/Style-shape-library

This documentation was supposed to be brought over to the website in #1116; now we have the following three pages:

- https://penrose.cs.cmu.edu/docs/ref/domain/syntax
- https://penrose.cs.cmu.edu/docs/ref/substance/syntax
- https://penrose.cs.cmu.edu/docs/ref/style/syntax

However, those pages are missing many items from the aforementioned wiki pages, as noted by @keenancrane:

> - strings
> - matrix/vector
> - type annotations
> - custom initialization value
> - math vs. text labels (`has math label` and `has text label`)
> - delete keyword
> - color
> - math functions
> - richer information about a couple shapes (e.g., `Path` and `Circle`)

> Also it’s weird that the page explaining the basic usage of Style is called “Syntax and Semantics” and is listed at the very bottom of the menu:

image

We need to move all this content from the wiki into the website. (Also, note that the above list is not necessarily exhaustive.)

### PR DESCRIPTION
**Pull request by liangyiliang**
# Description

Resolves #1314.

Moves applicable parts of the wiki documentations to the docs-site, including:
* `override` and `delete`
* strings
* matrix / vector
* type annotations
* some details of shapes
* colors
* computation functions
* labeling

Renamed `syntax and semantics` to `usage` with appropriate ordering.

Each page's outline is now a deep-outline, meaning that each sections and subsections will be shown in the outline. This makes navigation easier.

### CODE REVIEW

File: packages/docs-site/docs/ref/style/shapes-overview.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Shapes
+
+Style currently supports a variety of shapes, listed as sub-entries under this page. Note that the system is constantly evolving and this page may be out of date. Updated attributes for individual shapes can be found in the individual `Shapes.ts` files in the [`/shapes` subdirectory](https://github.com/penrose/penrose/tree/main/packages/core/src/shapes). In the attribute lists below, the `vec2` type describes a 2D vector `u`, whose scalar components can be accessed via `u[0]` and `u[1]`.
```
Comment (samestep) [inline packages/docs-site/docs/ref/style/shapes-overview.md:3]: I'd generally tend toward not linking to source code in user-facing docs. Is there some important info (for a Penrose user) behind this link? If so then we should document it here; if not then let's just remove the link.

File: packages/docs-site/docs/ref/style/shapes-overview.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+```
+forall Point x {
+   x.myCircle = Circle {
+      radius: 5
+      fillColor: rgba(1,0,0,1)
+   }
+}
+```
+
+Notice that this definition, the `center:` attribute of the circle is not set. Undefined parameters like these will be automatically optimized by Penrose.
```
Comment (samestep) [inline packages/docs-site/docs/ref/style/shapes-overview.md:28]: Might be a good idea to clarify this, since only some undefined parameters are optimized by default, whereas others are not.

File: packages/docs-site/docs/ref/domain/usage.md [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# Syntax
+# Usage
```
Comment (samestep) [inline packages/docs-site/docs/ref/domain/usage.md:1]: For consistency, since your updated header in the Style docs says "Style Usage":
```suggestion
# Domain Usage
```

File: packages/docs-site/docs/ref/substance/usage.md [hunk 4]
```diff
@@ -1,4 +1,4 @@
-# Syntax
+# Usage
```
Comment (samestep) [inline packages/docs-site/docs/ref/substance/usage.md:1]: Same comment here as in the Domain usage page:
```suggestion
# Substance Usage
```

File: packages/docs-site/docs/ref/style/usage.md [hunk 5]
```diff
@@ -55,7 +55,7 @@ where
 
 - `list_object_declarations` is a **semicolon**-separated list of object declarations, similar to the object declarations in the _substance_ schema. Each object declaration has syntax `type_name object_name`. The names declared in `list_object_declarations` are referred to as _style variables_.
 - `list_relations` is a **semicolon**-separated list of constraints (about objects in `list_object_declaration`) that must be satisfied in order for this style block to be triggered.
-- `list_body_expressions` are statements that represent the computational and graphical aspects of the diagrams that are triggered when this style block is triggered.
+- `list_body_expressions` is the body of this _style_ block, containing statements that represent the computational and graphical aspects of the diagrams that are triggered when this style block is triggered. Refer to [this section](usage#blody-body) for a detailed explanation of what may appear in the body of a _style_ block.
```
Comment (samestep) [inline packages/docs-site/docs/ref/style/usage.md:58]: Typo:
```suggestion
- `list_body_expressions` is the body of this _style_ block, containing statements that represent the computational and graphical aspects of the diagrams that are triggered when this style block is triggered. Refer to [this section](usage#block-body) for a detailed explanation of what may appear in the body of a _style_ block.
```

File: packages/docs-site/docs/ref/style/usage.md [hunk 6]
```diff
@@ -232,19 +234,34 @@ Set A, B
...
-### Style Block Body
+### Reserved Variables
+
+Within a _style_ block body, some variable names are reserved for metadata purposes:
 
-The body of a style block lists out computations, shape declarations, and constraint / objective definitions that define how to draw elements of a diagram.
+- `match_count` is an integer that refers to the number of times that this _style_ blocks will be triggered (or matched) in total; and
+- `match_id` is the 1-indexed ordinal of this current matching.
+
+These values can directly be read or overwritten within the style block body, or overwritten, if needed.
```
Comment (samestep) [inline packages/docs-site/docs/ref/style/usage.md:244]: This statement is a bit confusing since if says "overwritten" twice, I'm not quite sure what it means; could you rephrase it to be more clear?

File: packages/docs-site/docs/ref/style/usage.md [hunk 7]
```diff
@@ -254,24 +271,59 @@ For example, consider the following _style_ block:
...
+
+forall Set `A` {
+    override `A`.shape.r = 200
 }
 ```
 
-A full list of available shapes and their properties can be found in the Shape Library.
+the radius of the circle for every `Set` is `100`, except if the `Set` has name `A`, then the radius is `200`.
+
+Deletion of fields work similarly, with the `delete` keyword. This feature can be helpful for, e.g., removing visual elements for a subtype. For instance,
```
Comment (samestep) [inline packages/docs-site/docs/ref/style/usage.md:302]: Grammar nit:
```suggestion
Deletion of fields works similarly, with the `delete` keyword. This feature can be helpful for, e.g., removing visual elements for a subtype. For instance,
```

**Review by samestep — CHANGES_REQUESTED** [review]
Looks great, thanks so much for doing this! A few changes, and then I think we're good to merge :)

File: packages/docs-site/docs/ref/style/usage.md (hunk 6 above)
Comment (liangyiliang) [reply packages/docs-site/docs/ref/style/usage.md]: Typo - fixed

File: packages/docs-site/docs/ref/style/shapes-overview.md (hunk 2 above)
Comment (liangyiliang) [reply packages/docs-site/docs/ref/style/shapes-overview.md]: Done

File: packages/docs-site/docs/ref/style/shapes-overview.md (hunk 1 above)
Comment (liangyiliang) [reply packages/docs-site/docs/ref/style/shapes-overview.md]: Removed the link

**Review by samestep — APPROVED** [review]
Amazing! LGTM :rocket:
